src/routes/landing/components/Landing.js

```diff
@@ -24,19 +26,81 @@ export type PropsType = {|
   clearResourcesAndCache: () => void
 |}
 
-class Landing extends React.Component<PropsType> {
+export type LocationType = {| message: string |} | {|
+  longitude: number,
+  latitude: number
+|}
+
+type StateType = {|
+  proposeNearbyCities: boolean | null,
+  location: LocationType
+|}
+
+class Landing extends React.Component<PropsType, StateType> {
+  constructor (props: PropsType) {
+    super(props)
+    this.state = { proposeNearbyCities: null, location: { message: 'loading' } }
+  }
+
+  componentDidMount () {
+    this.initializeProposeCities()
+  }
+
+  initializeProposeCities = async () => {
+    const appSettings = new AppSettings()
+    const { proposeNearbyCities } = await appSettings.loadSettings()
+    this.setState({ proposeNearbyCities: proposeNearbyCities })
+
+    if (proposeNearbyCities) {
+      this.currentPosition()
+    }
+  }
+
+  currentPosition = () => {
+    this.setState({ location: { message: 'loading' } })
+    Geolocation.getCurrentPosition(
+      (position: GeolocationResponse) => {
+        this.setState({ location: {
+          longitude: position.coords.longitude,
+          latitude: position.coords.latitude
+        } })
+      },
+      (error: GeolocationError) => this.setLocationErrorMessage(error)
+      ,
+      { enableHighAccuracy: true, timeout: 50000, maximumAge: 3600000 }
+    )
+  }
+
+  setLocationErrorMessage = (error: GeolocationError) => {
+    if (error.code === 1) {
+      this.setState({ location: { message: 'noPermission' } })
+    } else if (error.code === 2) {
+      this.setState({ location: { message: 'notAvailable' } })
+    } else {
+      this.setState({ location: { message: 'timeout' } })
+    }
+  }
+
   navigateToDashboard = (cityModel: CityModel) => {
     const { navigateToDashboard, language } = this.props
     navigateToDashboard(cityModel.code, language)
   }
 
   render () {
     const { theme, cities, t, clearResourcesAndCache } = this.props
+    const { proposeNearbyCities } = this.state
+    const tryAgain = this.state.location.message === 'loading' ? null : this.currentPosition
 
-    return <Wrapper theme={theme}>
-      <Heading clearResourcesAndCache={clearResourcesAndCache} theme={theme} />
-      <FilterableCitySelector theme={theme} cities={cities} t={t} navigateToDashboard={this.navigateToDashboard} />
-    </Wrapper>
+    if (proposeNearbyCities !== null) {
+      return <Wrapper theme={theme}>
+        <Heading clearResourcesAndCache={clearResourcesAndCache} theme={theme} />
+        {/* $FlowFixMe Flow does not get that proposeNearbyCities is null */}
```

Comment:
proposeNearbyCities is not used in the following line. Not sure why flow should fail here.

Reply:
It was used by passing `{...this.state}`. However, the issue was fixed by explicitly passing it!